src/pages/gateway/fastly.md

```diff
@@ -0,0 +1,251 @@
+     - **Type** - **recv**
+     - **Priority** - **60**
+     - **Content**:
+
+        ```vcl
+        if ((req.request == "GET" || req.request == "HEAD") && (req.url.path ~ "/graphql" || req.url "^/api/(.*)") && req.url.qs ~ "query=" && req.http.x-commerce-bypass-fastly-cache == "true") {set req.http.graphql = "1";} else {unset req.http.graphql;} if ( req.url.path !~ "/graphql" && req.url !~ "^/api/(.*)" ) {
+
+        set req.http.Magento-Original-URL = req.url;
+
+        set req.url = querystring.regfilter(req.url,"^(utm_.*|gclid|gdftrk|_ga|mc_.*|trk_.*|dm_i|_ke|sc_.*|fbclid)$");
+        }
+        ...
+        ```
+
+   - Allows API Mesh to function as a backend:
+     - **Name** - api_mesh_backend
+     - **Type** - **init**
+     - **Priority** - **1**
+     - **Content**:
+  
+        ```vcl
+        backend F_graph_prod_adobe_io {
+            .always_use_host_header = true;
+            .between_bytes_timeout = 10s;
+            .connect_timeout = 1s;
+            .dynamic = true;
+            .first_byte_timeout = 15s;
+            .host = "graph.adobe.io";
+            .host_header = "graph.adobe.io";
+            .max_connections = 200;
+            .port = "443";
+            .share_key = "XXXXXXXXXXXXXXXX";
+            .ssl = true;
+            .ssl_cert_hostname = "graph.adobe.io";
+            .ssl_check_cert = always;
+            .ssl_sni_hostname = "graph.adobe.io";
+            .probe = {
+                .dummy = true;
+                .initial = 5;
+                .request = "HEAD / HTTP/1.1" "Host: graph.adobe.io" "Connection: close";
+                .threshold = 1;
+                .timeout = 2s;
+                .window = 5;
+            }
+        }
+        ```
+
+   - Enables the bypass header in API Mesh:
+     - **Name** - api_mesh_recv
+     - **Type** - **recv**
+     - **Priority** - **10**
+     - **Content**:
+
+        ```vcl
+        if(req.http.x-commerce-bypass-fastly-cache == "true") { return(pass); }
+        ```
+
+   - Cache miss - replace the `<mesh_id>` and `<mesh_api_key>` placeholders with the information from your mesh URL, which has the following structure: `https://graph.adobe.io/api/<meshId>/graphql?api_key=<your_apiKey>`:
+     - **Name** - api_mesh_miss
+     - **Type** - **miss**
+     - **Priority** - **60**
+     - **Content**:
+
+        ```vcl
+        if (req.url ~ "^/api/") {
+        set req.backend = F_graph_prod_adobe_io;
+        }
+        //apiMESH prod mapping
+        if (req.url ~ "^/api/<mesh_id>") {
+        set bereq.http.tenantuuid = "<mesh_id>";
+        set bereq.http.x-api-key = "<mesh_api_key>";
+        # Optionally add another environment
+        # }
+        # //apiMESH stage mapping
+        # if (req.url ~ "^/api/<mesh_id>") {
+        # set bereq.http.tenantuuid = "<mesh_id>";
+        # set bereq.http.x-api-key = "<mesh_api_key>";
+        # }
+        ```
+
+    The remaining snippets allow you to query your mesh URL without appending the `api_key`:
```

Comment:
Not a big deal, but I think this is only true for the `api_mesh_pass` snippet. It's essentially a duplicate of the `api_mesh_miss` snippet which routes to the appropriate backend and set the api key when querying a mesh.